[Vangmay Sachan] iP

Vangmay opened the pull request:
# Micheal Scott
> Regional Manager | Dunder Mifflin - Scranton branch | Worlds best boss

- Michael Scott is based on a character from my favourite TV show "The office"
- Its also easy to get tasks done when you have a boss holding you accountable. 

All you need to do is,

    1. download it from [here](https://github.com/Vangmay/ip/releases/tag/v0.2).
    2. run it from your command line using `java -jar ./ip.main.jar`
    3. add your tasks.
    4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Vangmay committed:
Level-0 commit

Vangmay committed:
Level-1

Vangmay committed:
Level-2

Vangmay committed:
Level-3 commit

Vangmay committed:
Level-4

Vangmay committed:
A-TextUiTesting

Vangmay committed:
A-Exceptions

Vangmay committed:
Level 5 and 6

Vangmay committed:
Add more exception handling

Vangmay committed:
Add testcases

Vangmay committed:
Edit MichaelScott.java

Vangmay committed:
create branch-level-8

Vangmay committed:
Add dateTime to Deadline

Vangmay committed:
Add LocalDateTime

Vangmay committed:
Create branch

Vangmay committed:
Create Storage, TaskList, Ui

Vangmay committed:
Create Parser

Vangmay committed:
Add more OOP

Vangmay committed:
Create branch-A-Packages

Vangmay committed:
merge gradle

Vangmay committed:
Add testCases

Vangmay committed:
create Jar

Vangmay committed:
Added javadoc comments

Vangmay committed:
Add feature find

Vangmay committed:
fix style

Vangmay committed:
Add MichaelScott to package

Vangmay committed:
Add package

Vangmay committed:
Add MichaelScott to package

Vangmay committed:
Add MichaelScott to package

Vangmay committed:
Add JavaDoc comments

Vangmay committed:
Edit Parser

Vangmay committed:
Edit parser

Vangmay committed:
add improvements

Vangmay committed:
Merge branch 'branch-A-JavaDoc'

Vangmay committed:
fix merge conflicts

Vangmay committed:
Merge Find command

Vangmay committed:
Fix parser bug

Vangmay committed:
change file structure

Vangmay committed:
Add load feature

Vangmay committed:
refactor storage

Vangmay committed:
refactor

Vangmay committed:
Edit messages

Vangmay committed:
Add checkstyle

Vangmay committed:
merge checkstyle branch

Vangmay committed:
Fix style

Vangmay committed:
add gui

Vangmay committed:
add gui

Vangmay committed:
delete main

Vangmay committed:
change ui

Vangmay committed:
merge Level 10

Vangmay committed:
add release

taggyhan commented on META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
```
Nice use of a manifest file! This helps ensure that the main class is clearly defined when running the application.

taggyhan commented on src/main/java/michaelscott/MichaelScott.java:
```diff
@@ -0,0 +1,72 @@
+        storage = new Storage(tasks);
+    }
+
+    public String getResponse(String input) throws MichaelScottException {
+        try {
+            CommandParser parser = new CommandParser();
+            Command c = parser.parse(input);
+            String result = c.execute(tasks);
+            this.commandType = c.getSimpleName();
+            storage.saveTasks(tasks.getTasks());
+            if (commandType.equals("ExitCommand")) {
+                return "Exit-Signal";
+            }
+            return result;
+        } catch (MichaelScottException e) {
+            throw e;
+        }
+    }
+
+    public String getCommandType() {
+        return commandType;
+    }
+
+    /**
+     * Runs the main execution loop of the chatbot.
+     * Handles user input, command execution, and output display.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isRunning = true;
+        while (isRunning) {
+            try {
+                String fullCommand = ui.readCommand();
+                Command cmd = parser.parse(fullCommand); //
+                String response = cmd.execute(tasks); //
+                ui.showResponse(response);
+                storage.saveTasks(tasks.getTasks()); //
+                isRunning = !cmd.isExit();
+            } catch (MichaelScottException e) {
+                Ui.showError(e.getMessage());
```
The method run() contains a few commented-out lines (e.g., //). It’s best to remove these if they are no longer necessary, to keep the code clean

taggyhan replied:
Overall, the code is clean and follows the Java naming conventions (e.g., camelCase for variables and PascalCase for class names). Nice job adhering to the Style Guide

taggyhan commented on src/main/java/michaelscott/utils/Ui.java:
```diff
@@ -0,0 +1,82 @@
+    /**
+     * Just shortening system.out.println.
+     */
+    public static void println(String sentence) {
+        System.out.println(sentence);
+    }
```
is this method necessary?

taggyhan reviewed (commented):
LGTM

yongqqqq commented on src/main/java/michaelscott/MichaelScott.java:
```diff
@@ -0,0 +1,72 @@
+     * Handles user input, command execution, and output display.
+     */
+    public void run() {
+        ui.showWelcome();
```
Maybe can name this as showWelcomeMessage instead?

yongqqqq commented on src/main/java/michaelscott/command/FindCommand.java:
```diff
@@ -0,0 +1,57 @@
+ * in their description and returns a list of matching tasks.
+ */
+public class FindCommand implements Command {
+    private final String keyWord;
```
maybe you can name this as keyword instead?

yongqqqq commented on src/main/java/michaelscott/utils/MichaelScottException.java:
```diff
@@ -0,0 +1,13 @@
+    /**
+     * Constructs an exception thrown in the MichaelScott chatbot.
+     */
+    public MichaelScottException(String errMessage) {
```
maybe can name as errorMessage instead?

yongqqqq commented on src/main/java/michaelscott/utils/Storage.java:
```diff
@@ -0,0 +1,83 @@
+     *
+     * @param todo the task list.
+     */
+    public void saveTasks(ArrayList<Task> todo) throws MichaelScottException {
```
maybe you can rename todo? As it may cause confusion with the Todo command u have

yongqqqq commented on src/main/java/michaelscott/command/DeleteCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @Override
+    public String getSimpleName() {
```
Good naming here

yongqqqq reviewed (commented):
Overall easy to understand with good naming!

Vangmay committed:
Change tasks.txt

Vangmay committed:
Add assertions

Added assertions where appropriate $'\n' Storage.java $'\n' Add assertions inside command package to ensure the parameters provided to commands are not null

Vangmay committed:
Fix code quality

delete useless code in MichaelScott.java
change variable naming in FindCommand.java
removed unrequired import in Dialogbox.java
changed michaelscott to final in Main.java
fix bug by adding finall clause in MainWindow.java
improve readability in Event.java, Task.java and TaskParser.java

Vangmay committed:
Modify MichaelScott

Vangmay committed:
Merge branch 'master' into branch-A-Assertions

Vangmay committed:
Merge pull request #1 from Vangmay/branch-A-Assertions

Add assertions

Vangmay committed:
Merge branch 'master' into branch-A-CodeQuality

Vangmay committed:
Merge pull request #2 from Vangmay/branch-A-CodeQuality

Branch a code quality

Vangmay committed:
Change tasks.txt

Vangmay committed:
Add support for period tasks

Inorder to add support for period tasks
it would be a good idea to add periodCommand.java
Edit the files in the parser so that it creates an appropriate
period event when reading from file

Vangmay committed:
Merge period tasks

Vangmay committed:
Remove time from period

Period is supposed to start at a date and end at a date

Let's
* Remove hh:mm from CommandParser, create a seperate data format for the Period task
* Add appropriate changes to the Period class
* Edit the taskParser such that when reading from file it doesn't require the hours and minutes

Vangmay committed:
Add personality

Vangmay committed:
Add user guide

Vangmay committed:
Fix User Guide

Vangmay committed:
Fix User Guide

Vangmay committed:
Add more tests

Vangmay committed:
Fix test

Vangmay committed:
Add style changes
